README.md

```diff
@@ -5,3 +5,183 @@ UCL API Main Repository
+    python3 python3-pip python3-virtualenv python-virtualenv \
+    unzip virtualenv libaio1 build-essential libpcre3 \
+    libpcre3-dev wget sed \
+    redis-server
+```
+
+### Start Redis
+```
+sudo service redis-server start
+```
+
+### Set up Oracle
+```
+# Oracle Version
+ORACLE_VERSION=12_2
+ORACLE_SO_VERSION=12.1
+
+# Oracle Instant Client download links
+ORACLE_INSTANTCLIENT_BASIC_URL=https://s3.eu-west-2.amazonaws.com/uclapi-static/instantclient-basic-linux.x64-12.2.0.1.0.zip
+ORACLE_INSTANTCLIENT_SDK_URL=https://s3.eu-west-2.amazonaws.com/uclapi-static/instantclient-sdk-linux.x64-12.2.0.1.0.zip
+
+sudo mkdir /opt/oracle
+
+rm -rf /opt/oracle/instantclient_$ORACLE_VERSION
+
+wget -O instantclient.zip "$ORACLE_INSTANTCLIENT_BASIC_URL" 
+wget -O instantclientsdk.zip "$ORACLE_INSTANTCLIENT_SDK_URL"
+
+sudo unzip -d/opt/oracle temp/instantclient.zip
+sudo unzip -d/opt/oracle temp/instantclientsdk.zip
+
+export ORACLE_HOME=/opt/oracle/instantclient_$ORACLE_VERSION
+
+pushd $ORACLE_HOME
+
+sudo ln -s libclntsh.so.$ORACLE_SO_VERSION libclntsh.so
+sudo ln -s libocci.so.$ORACLE_SO_VERSION libocci.so
+sudo ln -s libclntshcore.so.$ORACLE_SO_VERSION libclntshcore.so
+
+export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$ORACLE_HOME
+sudo ldconfig
+
+# Add ORACLE_HOME to /etc/environment
+grep -q -F "ORACLE_HOME=/opt/oracle/instantclient_$ORACLE_VERSION" /etc/environment || sudo echo "ORACLE_HOME=/opt/oracle/instantclient_$ORACLE_VERSION" >> /etc/environment
+
+# Add ld path to /etc/ld.so.conf.d
+sudo echo "/opt/oracle/instantclient_$ORACLE_VERSION" > /etc/ld.so.conf.d/oracle.conf
+sudo ldconfig
+
+popd
+```
+
+### Download the UCL API Source Code
+```
+git clone https://github.com/uclapi/uclapi
+```
+
+### Fetch Fake Shibboleth
+```
+git clone https://github.com/uclapi/fakeshibboleth
+```
+
+### Set up a virtual environment for UCL API
+```
+pushd uclapi/backend/uclapi
+virtualenv --python=python3 venv
+. venv/bin/activate
+pip install -r requirements.txt
+deactivate
+popd
+```
+
+If this builds and installs then your Oracle configuration is also working. If it fails, try restarting your shell to pick up the new environment variables set up in the Oracle set up above. If that doesn't work, set `ORACLE_HOME` to the instantclient directory.
+
+### Set up a virtual environment for Fake Shibboleth
+```
+pushd fakeshibboleth
+virtualenv --python=python3 venv
+. venv/bin/activate
+pip install -r requirements.txt
+deactivate
+popd
+```
+
+### Install PostgreSQL
+Setting this up will vary based on your operating system. It is perfectly possible to just use the Windows version of Postgres and run it under Windows whilst running the rest of the code under Linux. If you are working on Linux or macOS directly then you should install PostgreSQL and then reset the `postgres` account password to one you know and can save in the .env later.
+
+### Configure the environment variables in .env
+Firstly, copy uclapi/backend/uclapi/.env.example to uclapi/backend/uclapi/.env.
```

Comment:
just consistency, so no not much. 

But if you're making changes might as well add this.